EKF/control.cpp

```diff
@@ -394,8 +394,21 @@ void Ekf::controlOpticalFlowFusion()
 
 		// optical flow fusion mode selection logic
 		bool flow_quality_ok = _flow_sample_delayed.quality >= _params.flow_qual_min_init;
+
+		if (!_control_status.flags.in_air) {
+			_time_takeoff = 0;
+		} else if (_time_takeoff == 0) {
+			_time_takeoff = _imu_sample_delayed.time_us;
+		}
+
 		if (!flow_quality_ok) {
-			_time_bad_flow_qual = _imu_sample_delayed.time_us;
+			if ((!_control_status.flags.in_air || (_control_status.flags.in_air && _imu_sample_delayed.time_us - _time_takeoff < (uint64_t)5E6))
+				&& _params.flow_qual_min_check) {
+				flow_quality_ok = true;
+				_time_bad_flow_qual = 0;
+			} else {
+				_time_bad_flow_qual = _imu_sample_delayed.time_us;
+			}
```

Comment:
Could you add some comment for these code for better understanding?

Reply:
ok, done in commit "EKF:add some comment and simple code"